### PR TITLE
Implements Brood of Cockroaches

### PR DESCRIPTION
**Pull request by mathieupouedras**

### CODE REVIEW

File: Mage.Sets/src/mage/cards/b/BroodOfCockroaches.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        AtTheBeginOfNextEndStepDelayedTriggeredAbility delayedLifeLost =
+                new AtTheBeginOfNextEndStepDelayedTriggeredAbility(new LoseLifeSourceControllerEffect(1));
+        game.addDelayedTriggeredAbility(delayedLifeLost, source);
+
+        Effect effect = new ReturnToHandTargetEffect();
+        effect.setText("return Brood of Cockroaches to your hand.");
```
Comment (LevelX2) [inline Mage.Sets/src/mage/cards/b/BroodOfCockroaches.java]: Does {this} instead of the name of the card not work in this case?

**Comment by mathieupouedras** [comment]
my **test.dck**

NAME:testz
1 [TMP:320] Mountain
1 [LRW:297] Mountain
1 [HOU:194] Swamp
1 [CHK:295] Swamp
1 [ROE:238] Swamp
11 [10E:232] Shock
1 [6ED:345] Mountain
1 [MIR:304] Mountain
1 [SHM:291] Swamp
1 [M11:240] Swamp
1 [ISD:261] Mountain
1 [M14:242] Mountain
1 [6ED:339] Swamp
1 [AKH:266] Mountain
1 [ALA:238] Swamp
1 [MRD:298] Swamp
11 [VIS:3] Brood of Cockroaches
1 [HOU:188] Mountain
1 [AKH:264] Mountain
1 [LEB:294] Swamp
LAYOUT MAIN:(1,3)(CMC,false,100)|([TMP:320],[AKH:264],[6ED:345],[MIR:304],[ISD:261],[M14:242],[AKH:266],[LRW:297],[HOU:188],[ALA:238],[HOU:194],[LEB:294],[6ED:339],[M11:240],[ROE:238],[CHK:295],[SHM:291],[MRD:298])([10E:232],[10E:232],[10E:232],[10E:232],[10E:232],[10E:232],[10E:232],[10E:232],[10E:232],[10E:232],[10E:232])([VIS:3],[VIS:3],[VIS:3],[VIS:3],[VIS:3],[VIS:3],[VIS:3],[VIS:3],[VIS:3],[VIS:3],[VIS:3])
LAYOUT SIDEBOARD:(0,0)(COLOR,true,100)|

**Comment by LevelX2** [comment]
Thanks for your contribution.

**Comment by Zzooouhh** [comment]
How does this function if it dies under another player's control? The Cockroaches owner (i.e. the player whose graveyard it lands in) should be the one losing life and resolving the return to hand effect.

**Comment by mathieupouedras** [comment]
Not sure how it works in that scenario. I will test it and fix what has to be fixed for this test case

**Comment by mathieupouedras** [comment]
Let me know if you can reopen this or should i create a new issue

**Pull request merged**